### PR TITLE
Improvement/test namnx228

### PR DESCRIPTION
**Pull request by namnx228**

### CODE REVIEW

File: vm-setup/roles/libvirt/tasks/install_setup_tasks.yml [hunk 1]
```diff
@@ -1,29 +1,7 @@
...
-    - ansible_facts['distribution'] == "Ubuntu"
-  package:
-    name: "{{ ubuntu_libvirt_packages }}"
-    state: present
-  become: true
-
+- name: Install libvirt dependencies
+  block:
+    - include_tasks: install_package_tasks.yml
+  when: libvirt_action == "install_package"
```
Comment (maelk) [inline vm-setup/roles/libvirt/tasks/install_setup_tasks.yml]: Is this needed if it runs in 01 ? and starting libvirtd could be considered as an installation step

File: vm-setup/roles/libvirt/tasks/main.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
     - include_tasks: install_setup_tasks.yml
+  when:  libvirt_action == "install_package"
+
+- name: libvirt role setup tasks
+  block:
     - include_tasks: network_setup_tasks.yml
     - include_tasks: vm_setup_tasks.yml
       when: vm_platform == "libvirt"
-  when: libvirt_action == "setup"
+  when: libvirt_action == "setup"
```
Comment (maelk) [inline vm-setup/roles/libvirt/tasks/main.yml]: unneeded space

**Review by maelk — COMMENTED** [review]
Let's discuss about the use of roles

File: vm-setup/install-package-playbook.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+- name: Install packages needed for the Dev-env
+  hosts: virthost
+  connection: local
+  gather_facts: true
+  vars_files:
+    - vars/input.yml
```
Comment (maelk) [inline vm-setup/install-package-playbook.yml]: please use more explicit file naming

File: vm-setup/install-package-playbook.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  tasks:
+    - import_role:
+        name: packages_installation
+  vars: 
+    install: "{{ input.packages_installation }}"
```
Comment (maelk) [inline vm-setup/install-package-playbook.yml]: you can remove one level here

File: vm-setup/roles/common/tasks/main.yml [hunk 5]
```diff
@@ -1,70 +1,72 @@
...
-              - 1024
-              - 65535
-            domain: "{{ cluster_domain }}"
-            dns:
-              hosts: "{{dns_extrahosts | default([])}}"
-              forwarders:
-                - domain: "apps.{{ cluster_domain }}"
-                  addr: "127.0.0.1"
+- name: Setup VMs
+  when: libvirt_action != "install_package"
```
Comment (maelk) [inline vm-setup/roles/common/tasks/main.yml]: remove this condition, it is not needed

File: vm-setup/roles/packages_installation/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      block: 
+        - name: Install packages on rhel7
+          when: ansible_lsb.major_release|int == 7
+          package:
+             name: "{{ install.centos.package +  install.centos.libvirt.rhel7 }}"
+             state: present
+        - name: Install packages on rhel8
+          when: ansible_lsb.major_release|int == 8
+          package:
+             name: "{{ install.centos.package + install.centos.libvirt.rhel7 }}"
```
Comment (maelk) [inline vm-setup/roles/packages_installation/tasks/main.yml]: rhel8 here

File: vm-setup/vars/input.yml [hunk 7]
```diff
@@ -0,0 +1,96 @@
+input:
```
Comment (maelk) [inline vm-setup/vars/input.yml]: you can remove the input

**Review by maelk — CHANGES_REQUESTED** [review]

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

File: vm-setup/roles/common/tasks/main.yml [hunk 8]
```diff
@@ -1,70 +1,71 @@
...
-              - 1024
-              - 65535
-            domain: "{{ cluster_domain }}"
-            dns:
-              hosts: "{{dns_extrahosts | default([])}}"
-              forwarders:
-                - domain: "apps.{{ cluster_domain }}"
-                  addr: "127.0.0.1"
+- name: Setup VMs
+  block:
```
Comment (maelk) [inline vm-setup/roles/common/tasks/main.yml]: why a block here ?

Comment (maelk) [reply vm-setup/roles/common/tasks/main.yml]: This file should not be touched, right?

**Review by maelk — CHANGES_REQUESTED** [review]

File: vm-setup/vars/packages_installation.yml [hunk 9]
```diff
@@ -0,0 +1,95 @@
+packages_installation:
+  ubuntu:
+    package:
```
Comment (maelk) [inline vm-setup/vars/packages_installation.yml]: packages

File: vm-setup/vars/packages_installation.yml [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+     - virtualbmc 
+     - python-ironicclient 
+     - python-ironic-inspector-client 
+     - lxml 
+     - netaddr 
+     - requests 
+     - setuptools 
+     - libvirt-python 
+  centos:
+    package:
```
Comment (maelk) [inline vm-setup/vars/packages_installation.yml]: packages

File: vm-setup/roles/packages_installation/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,36 @@
+- name: Install packages on Ubuntu
+  block:
+    - name: Install packages using standard package manager
+      package:
+        name: "{{ package.ubuntu.package +  package.ubuntu.libvirt }}"
```
Comment (maelk) [inline vm-setup/roles/packages_installation/tasks/main.yml]: packages (*3)

File: vm-setup/roles/packages_installation/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+        state: present
+      become: true
+    - name: Install packages using pip
+      pip:
+        name: "{{ package.ubuntu.pip }}"
```
Comment (maelk) [inline vm-setup/roles/packages_installation/tasks/main.yml]: same and further in the file

File: vm-setup/install-package-playbook.yml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  hosts: virthost
+  connection: local
+  gather_facts: true
+  vars_files:
+    - vars/packages_installation.yml
+  tasks:
+    - import_role:
+        name: packages_installation
+  vars: 
+    package: "{{ packages_installation }}"
```
Comment (maelk) [inline vm-setup/install-package-playbook.yml]: packages

**Comment by maelk** [comment]
we will not continue further in this direction for now

**Pull request closed without merging**